core/payment/src/api/allocations.rs

```diff
@@ -54,19 +57,27 @@ async fn create_allocation(
     };
     match async move { Ok(bus::service(LOCAL_SERVICE).send(validate_msg).await??) }.await {
         Ok(true) => {}
-        Ok(false) => return response::bad_request(&"Insufficient funds to make allocation"),
+        Ok(false) => return response::bad_request(&"Insufficient funds to make allocation. Release all existing allocations to unlock the funds via `yagna payment release-allocations`"),
         Err(Error::Rpc(RpcMessageError::ValidateAllocation(
             ValidateAllocationError::AccountNotRegistered,
         ))) => return response::bad_request(&"Account not registered"),
         Err(e) => return response::server_error(&e),
     }
 
-    let dao: AllocationDao = db.as_dao();
+    let db_ = db.clone();
+    let dao = db.as_dao::<AllocationDao>();
+
     match async move {
         let allocation_id = dao
             .create(allocation, node_id, payment_platform, address)
             .await?;
-        Ok(dao.get(allocation_id, node_id).await?)
+        match dao.get(allocation_id, node_id).await? {
+            None => Ok(None),
+            Some(allocation) => {
+                release_allocation_after(db_, allocation.clone(), Some(node_id.clone())).await;
```

Comment:
Why is this function not on the "Data Access Object"? this would remove the ugly db_ clone

```suggestion
                dao.release_allocation_after(allocation.clone(), Some(node_id.clone())).await;
```